app/utils/callback/lava.py

```diff
@@ -85,291 +69,249 @@
+
+    def __init__(self, job_data, definition_meta, lab_name,
+                 base_path):
+        self._job_data = job_data
+        self.base_path = base_path
+        self.results = self._prepare_results(job_data["results"])
+        self.meta = self._prepare_meta(job_data, definition_meta, lab_name)
+        self.definition = yaml.load(job_data["definition"],
+                                    Loader=yaml.CLoader)
+        self.log = self._prepare_log(job_data["log"])
+
+    def _get_lava_job_meta(self, boot_meta):
+        """Parse the job meta-data from LAVA
+
+        :param meta: The boot meta-data.
+        :type meta: dictionary
+        :param boot_meta: The boot and auto_login meta-data
+               from the LAVA v2 job.
+        :type boot_meta: dictionary
+        """
+        meta = {}
+        if boot_meta.get("error_type") == "Infrastructure":
+            meta[models.BOOT_RESULT_KEY] = "UNKNOWN"
+        return meta
+
+    def _get_lava_boot_meta(self, boot_meta):
+        """Parse the boot and login meta-data from LAVA
+
+        :param boot_meta: The boot and auto_login meta-data
+               from the LAVA v2 job.
+        :type boot_meta: dictionary
+        :return meta: The boot meta-data dict
+        """
+        meta = {}
+        meta[models.BOOT_TIME_KEY] = boot_meta["duration"]
+        extra = boot_meta.get("extra", None)
+        if extra is None:
+            return
+        kernel_messages = []
+        for e in extra:
+            fail = e.get("fail", None)
+            if not fail:
+                continue
+            if isinstance(fail, str):
+                kernel_messages.append(fail)
+            else:
+                for msg in (f.get("message", None) for f in fail):
+                    if msg:
+                        kernel_messages.append(msg)
+        if kernel_messages:
+            meta[models.BOOT_WARNINGS_KEY] = len(kernel_messages)
+        return meta
+
+    def _get_lava_bootloader_meta(self, bl_meta):
+        """Parse the bootloader meta-data from LAVA
+
+        :param bl_meta: The bootloader meta-data from the LAVA v2 job.
+        :type bl_meta: dictionary
+        :return The boot meta-data
+        """
+        meta = {}
+        extra = bl_meta.get("extra", None)
+        if extra is None:
+            return
+        for e in extra:
+            for k, v in e.iteritems():
+                meta_key = BL_META_MAP.get(k, None)
+                if meta_key:
+                    meta[meta_key] = v
+        return meta
+
+    def _get_directory_path(self, meta):
+        """Create the dir_path from LAVA metadata
+
+        Update the metadata with the storage path of the artifacts.
+        If possible, use the file_server_resource from the metadata.
+
+        :param meta: The boot meta-data.
+        :type meta: dictionary
+        """
+
+        file_server_resource = meta.get(models.FILE_SERVER_RESOURCE_KEY)
+        if file_server_resource:
```

Comment:
Out of interest, is this not always the case now?

Maybe something to simplify in a follow-up PR if we can prove that "file server resource" is always provided.